src/cloudinary-image.component.spec.ts

```diff
@@ -524,4 +526,155 @@ describe('CloudinaryImage', () => {
+      const count = async () => document.querySelectorAll('.startWindow').length;
+      const scrollDown = async () => {
+        document.querySelector('.endWindow')
+          .scrollIntoView({ behavior: 'smooth', block: 'end', inline: 'end' });
+      }
+
+      let preCount = 0;
+      let postCount = 0;
+      do {
+        preCount = await count();
+        await scrollDown();
+        await wait(delay);
+        postCount = await count();
+      } while (postCount > preCount);
+      await wait(delay);
+
+      const img = des[3].children[0].nativeElement as HTMLImageElement;
+      expect(img.hasAttribute('src')).toBe(true);
+      expect(img.attributes.getNamedItem('src').value).toEqual(jasmine.stringMatching('image/upload/bear'));
+    });
+  });
+  describe('lazy load image', async () => {
+    @Component({
+      template: `
+          <div class="startWindow"><cl-image loading="lazy" width="300" public-id="bear"></cl-image></div>
+          <div style="margin-top: 300px"><cl-image loading="lazy" width="300" public-id="bear"></cl-image></div>
+          <div style="margin-top: 300px">
+              <cl-image loading="lazy" width="300" public-id="bear">
+                  <cl-placeholder type="pixelate"></cl-placeholder>
+              </cl-image>
+          </div>
+          <div style="margin-top: 300px"><cl-image loading="lazy" width="300" public-id="bear"></cl-image></div>
+          <div class="endWindow" style="margin-top: 300px"><cl-image loading="lazy" width="300" public-id="bear"></cl-image></div>`
+    })
+    class TestComponent {}
+
+    let fixture: ComponentFixture<TestComponent>;
+    let des: DebugElement[];  // the elements w/ the directive
+    let testLocalCloudinary: Cloudinary = new Cloudinary(require('cloudinary-core'),
+      { cloud_name: '@@fake_angular2_sdk@@', client_hints: true } as CloudinaryConfiguration);
+    beforeEach(() => {
+      fixture = TestBed.configureTestingModule({
+        declarations: [CloudinaryTransformationDirective, CloudinaryImage, TestComponent, LazyLoadDirective, CloudinaryPlaceHolder],
+        providers: [{ provide: Cloudinary, useValue: testLocalCloudinary }]
+      }).createComponent(TestComponent);
+
+      fixture.detectChanges(); // initial binding
+      // all elements with an attached CloudinaryImage
+      des = fixture.debugElement.queryAll(By.directive(CloudinaryPlaceHolder));
+    });
+    it('should load placeholder eagerly', fakeAsync(() => {
+      tick();
+      fixture.detectChanges();
+      const img = des[0].children[0].nativeElement as HTMLImageElement;
+      expect(img.attributes.getNamedItem('src').value).toEqual(jasmine.stringMatching('image/upload/e_pixelate,f_auto,q_1/bear'));
+    }));
+  });
+  describe('placeholder', () => {
+    @Component({
+      template: `<cl-image public-id="bear" width="300" crop="fit">
+          <cl-placeholder type="blur"></cl-placeholder>
+      </cl-image>`
+    })
+    class TestComponent {}
```

Comment:
@strausr , missing:
1. Test with no type
2. Test with all other types to make sure we get the relevant transformation
3. Test with lazy on the parent 
4. Can we test that the replacement took place?

Reply:
Done. Except for 3. Testing the lazy on the parent is redundant since the first resource will always load (always in viewport)